instrumentation/internal/internal-application-logger/bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/logging/ApplicationLoggerBridge.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.bootstrap.logging;
+
+import io.opentelemetry.javaagent.bootstrap.InternalLogger;
+import java.util.concurrent.atomic.AtomicReference;
+
+public abstract class ApplicationLoggerBridge {
+
+  private static final AtomicReference<ApplicationLoggerBridge> applicationLoggerBridge =
+      new AtomicReference<>();
+
+  public static void set(ApplicationLoggerBridge bridge) {
+    applicationLoggerBridge.compareAndSet(null, bridge);
+  }
+
+  public static void installApplicationLogger(InternalLogger.Factory applicationLoggerFactory) {
+    ApplicationLoggerBridge bridge = applicationLoggerBridge.get();
+    if (bridge != null) {
```

Comment:
can we throw exception on bridge `null`, or is that valid in some cases?

Reply:
This actually should never happen; a `null` here would mean a misconstructed agent that does not have the `LoggingCustomizer` implementation, but still runs the instrumentation.